File: installer/src/test/java/com/teamscale/profiler/installer/EnvironmentMapTest.java
package com.teamscale.profiler.installer;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class EnvironmentMapTest {

	@Test
	void testQuoting() {
		assertThat(new JvmEnvironmentMap("V", "a b").getSystemdString()).isEqualTo("\"V=a b\"");
		assertThat(new JvmEnvironmentMap("V", "a b").getEtcEnvironmentLinesList()).containsExactlyInAnyOrder(
				"V=\"a b\"");
		assertThat(new JvmEnvironmentMap("V", "a b").getEnvironmentVariableMap()).containsEntry("V", "\"a b\"");
	}

	@Test
	void testMultipleEntries() {
		assertThat(new JvmEnvironmentMap("V", "a", "E", "b").getSystemdString()).isEqualTo("E=b V=a");
		assertThat(new JvmEnvironmentMap("V", "a", "E", "b").getEtcEnvironmentLinesList()).containsExactlyInAnyOrder(
				"E=b", "V=a");
	}

}

File: installer/src/test/kotlin/com/teamscale/profiler/installer/EnvironmentMapTest.kt
package com.teamscale.profiler.installer

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

internal class EnvironmentMapTest {
	@Test
	fun testQuoting() {
		Assertions.assertThat(JvmEnvironmentMap("V", "a b").systemdString).isEqualTo("\"V=a b\"")
		Assertions.assertThat(JvmEnvironmentMap("V", "a b").etcEnvironmentLinesList).containsExactlyInAnyOrder(
			"V=\"a b\""
		)
		Assertions.assertThat(JvmEnvironmentMap("V", "a b").environmentVariableMap).containsEntry("V", "\"a b\"")
	}

	@Test
	fun testMultipleEntries() {
		Assertions.assertThat(JvmEnvironmentMap("V", "a", "E", "b").systemdString).isEqualTo("E=b V=a")
		Assertions.assertThat(JvmEnvironmentMap("V", "a", "E", "b").etcEnvironmentLinesList).containsExactlyInAnyOrder(
			"E=b", "V=a"
		)
	}
}